core/server/common/src/main/java/alluxio/underfs/UfsManager.java

```diff
@@ -43,17 +44,18 @@ public UfsClient(Supplier<UnderFileSystem> ufsSupplier, AlluxioURI ufsMountPoint
       mUfsMountPointUri = Preconditions.checkNotNull(ufsMountPointUri, "ufsMountPointUri is null");
       mCounter = MetricsSystem.counter(
           String.format("UfsSessionCount-Ufs:%s", MetricsSystem.escape(mUfsMountPointUri)));
+      mUfs = new AtomicReference<>();
     }
 
     /**
      * @return the UFS instance
      */
-    public synchronized CloseableResource<UnderFileSystem> acquireUfsResource() {
-      if (mUfs == null) {
-        mUfs = mUfsSupplier.get();
+    public CloseableResource<UnderFileSystem> acquireUfsResource() {
+      if (mUfs.get() == null) {
+        mUfs.compareAndSet(null, mUfsSupplier.get());
```

Comment:
unsafe here, better to check the result of cas operation and close resource accordingly.
```
UnderFileSystem ufs = mUfsSupplier.get();
if (!mUfs.compareAndSet(null, ufs)) ufs.close();
```

Reply:
Added close. Thanks!